artifactory/commands/generic/download.go

```diff
@@ -96,14 +97,28 @@ func (dc *DownloadCommand) Run() error {
 		downloadParamsArray = append(downloadParamsArray, downParams)
 	}
 	// Perform download.
-	filesInfo, totalExpected, err := servicesManager.DownloadFiles(downloadParamsArray...)
+	// In case of build-info collection/sync-deletes operation/a detailed summary is required, we use the download service which provides results file reader,
```

Comment:
I suggeast adding the following comment here:

// If the 'details summary' was requested, then the reader should not be closed now. It will be closed after it will be used to generate the summary.